javascript/javascript1/week4/index.js

```diff
@@ -0,0 +1,205 @@
+    recognition.stop();
+  };
+
+  recognition.onresult = function (event) {
+    let msg = "";
+    let transcript = event.results[0][0].transcript;
+
+    if (
+      /hi my name is/gi.test(transcript) ||
+      /my name is/gi.test(transcript) ||
+      /i am/gi.test(transcript)
+    ) {
+      personalInformation.name = transcript.split(" ").pop();
+      msg = `hi ${personalInformation.name}`;
+      console.log(msg);
+    } else if (
+      /what is my name/gi.test(transcript) &&
+      personalInformation.name === ""
+    ) {
+      msg = `You did not add name, Add you name please`;
+    } else if (
+      /what is my name/gi.test(transcript) &&
+      personalInformation.name !== ""
+    ) {
+      msg = `Your name is ${personalInformation.name}`;
+    } else if (
+      /my age is/gi.test(transcript) ||
+      /old/gi.test(transcript) ||
+      /years/gi.test(transcript) ||
+      /my old is/gi.test(transcript)
+    ) {
+      personalInformation.age = transcript.split(" ").pop();
+      console.log(personalInformation);
+      msg = `Your age is ${personalInformation.age} years old`;
+    } else if (
+      /what is my age/gi.test(transcript) &&
+      personalInformation.age === ""
+    ) {
+      msg = `You did not add age, Add you age please`;
+    } else if (
+      /what is my age/gi.test(transcript) &&
+      personalInformation.age !== ""
+    ) {
+      msg = `Your age is ${personalInformation.age}`;
+    } else if (/todo/gi.test(transcript) || /to do/gi.test(transcript)) {
+      msg = `you have ${todos.length} tasks on your todo,
+          if you want to add task use add,
+          if you want to see all your tasks use check,
+          if you want to remove something from todo use remove`;
+    } else if (/add to todo/gi.test(transcript)) {
+      msg = `What do you want to add todo`;
+    } else if (
+      /add/gi.test(transcript) ||
+      /add to do/gi.test(transcript) ||
+      / add to todo/gi.test(transcript)
+    ) {
+      transcript = transcript.split(" ");
+      console.log(transcript);
+      transcript.splice(transcript.indexOf("add"), 1);
+      transcript = transcript.join(" ");
+      todos.push(transcript);
+      console.log(todos);
+      msg = `${transcript} added to your todo`;
+    } else if (
+      /what is on my todo/gi.test(transcript) ||
+      /what i have in my todo/gi.test(transcript) ||
+      /check todo/gi.test(transcript) ||
+      /check/gi.test(transcript)
+    ) {
+      for (let i = 0; i < todos.length; i++) {
+        msg = todos[i];
+        console.log(msg);
+        textToAudio(msg);
+        msg = "";
+      }
+    } else if (/remove/gi.test(transcript)) {
+      transcript = transcript.split(" ").splice(transcript.indexOf("add", 1));
+      console.log(transcript);
+      let removeTodoElment = todos.splice(todos.indexOf(transcript), 1);
+      console.log(todos);
+      msg = ` Removed ${removeTodoElment} from your todo`;
```

Comment:
great logging the `removeTodoElment`, but is the `todos` array actually updated?